### PR TITLE
Introduce children function support

### PR DESCRIPTION
**Pull request by maxmalov**
I've found that children function support allows to pass sticky state to children components in a very handy way. This approach can be used instead of introducing another wrapper component that will handle state change with `onStateChange` and pass it to children components

### CODE REVIEW

**Comment by yahoocla** [comment]
Thank you for submitting this pull request, however I do not see a valid CLA on file for you. Before we can merge this request please visit https://yahoocla.herokuapp.com/ and agree to the terms. Thanks! :smile:

**Comment by redonkulus** [comment]
@maxmalov have you looked into the React Hooks work that came out recently? Would this help as well?

**Comment by maxmalov** [comment]
I'm not sure I follow you since I need to do extra state management even with hooks API, like I need to create one more component

```js
function StickyWrapper() {
  const [status, setStatus] = useState();
  return (
    <Sticky onStateChange={setStatus}>
      <MyComponentToStick status={status} />
    </Sticky>
}
```

On the other hand, children function exposes existing state to children components, so I don't need to create any wrapper components and use it directly in the application code

```js
<Sticky>
  {status => <MyComponentToStick status={status} />}
</Sticky>
```

File: src/Sticky.jsx [hunk 1]
```diff
@@ -380,10 +380,12 @@ class Sticky extends Component {
             [this.props.releasedClass]: this.state.status === STATUS_RELEASED
         })
 
+        var children = this.props.children;
+
         return (
             <div ref={(outer) => { this.outerElement = outer; }} className={outerClasses} style={outerStyle}>
                 <div ref={(inner) => { this.innerElement = inner; }} className='sticky-inner-wrapper' style={innerStyle}>
-                    {this.props.children}
+                    {typeof children === 'function' ? children({ status: this.state.status }) : children}
```
Comment (redonkulus) [inline src/Sticky.jsx]: Would it make sense to pass the `state` into the children instead of the `status` only?

Comment (maxmalov) [reply src/Sticky.jsx]: Not sure other things from the component state should be available to children, also `onStateChange` passes `status` only too https://github.com/yahoo/react-stickynode/blob/master/src/Sticky.jsx#L290-L293

**Comment by redonkulus** [comment]
react-stickynode 2.1.0 published 🎉

**Pull request merged**